Say, if `body_string` is an integer, and not a string.

Before:

```
bootstrap.js:5 Error importing `index.js`: Error: Malformed signing payload, check docs for details.
```

After:

```
bootstrap.js:5 Error importing `index.js`: Error: invalid type: integer `1`, expected a string at line 1 column 48
    at Module.__wbg_new_6cd733917efb279a (hp_admin_key_manager.js:225)
    at __wbg_new_6cd733917efb279a (bootstrap.js:77)
    at js_sys::Error::new::hed01abee6b29d057 (:8080/wasm-function[359]:0x317f3)
    at hp_admin_key_manager::util::into_js_error::h6e88c96d4311960b (:8080/wasm-function[89]:0x29b7a)
    at hp_admin_key_manager::HpAdminKeypair::sign::h93b2b8c86d6b9ada (:8080/wasm-function[42]:0x208a9)
    at hpadminkeypair_sign (:8080/wasm-function[128]:0x2d119)
    at HpAdminKeypair.sign (hp_admin_key_manager.js:176)
    at eval (index.js:17)
    at Module../index.js (0.bootstrap.js:34)
    at __webpack_require__ (bootstrap.js:105)
```

Compare which output has useful information for debugging.

--- a/client/CONTRIBUTING.md
+++ b/client/CONTRIBUTING.md
@@ -20,7 +20,7 @@ cargo test
 ### Wasm-pack unit tests
 Test interaction with created Wasm package in selected environments:
 ```
-wasm-pack --firefox
+wasm-pack test --firefox
 ```
 Must specify at least one of `--node`, `--chrome`, `--firefox`, or `--safari`.
 

--- a/client/README.md
+++ b/client/README.md
@@ -20,7 +20,7 @@ let kp = new HpAdminKeypair(HC_PUBLIC_KEY, EMAIL, PASSWORD);
 const payload = {
     method: "get",
     uri: "/someuri",
-    body_string: ""
+    body: ""
 }
 
 console.log(kp.sign(payload));

--- a/client/src/lib.rs
+++ b/client/src/lib.rs
@@ -27,7 +27,7 @@ impl HpAdminKeypair {
         email: String,
         password: String,
     ) -> Fallible<HpAdminKeypair> {
-        // console_error_panic_hook::set_once();
+        console_error_panic_hook::set_once();
         let keypair = new_inner(hc_public_key_string, email, password)?;
         Ok(Self(keypair))
     }
@@ -69,7 +69,7 @@ fn new_inner(
     let hc_public_key = PublicKey::from_bytes(&hc_public_key_bytes)
         .map_err(into_js_error)?;
     let keypair = admin_keypair_from(hc_public_key, &email, &password)
-       .map_err(|e| JsValue::from(e.to_string()))?;
+        .map_err(|e| JsValue::from(e.to_string()))?;
     Ok(keypair)
 }
 
@@ -77,7 +77,7 @@ fn parse_payload(payload: &JsValue) -> Fallible<Vec<u8>> {
     let payload_struct: Payload = payload.into_serde()
         .map_err(into_js_error)?;
     Ok(serde_json::to_vec(&payload_struct)
-       .map_err(into_js_error)?)
+        .map_err(into_js_error)?)
 }
 
 #[cfg(test)]
@@ -89,7 +89,7 @@ mod tests {
     const EMAIL: &str = "pj@abba.pl";
     const PASSWORD: &str = "abba";
     const EXPECTED_SIGNATURE: &str =
-        "viqqtbfbhTtAarkJjzOgO4cu4MFgGnshYHMjV3nITem01js9lTq0bG2Hwn9rXJi6xYVqOnq2NcosEMcRZ1CAAA";
+        "H/7BGNKb7ICPR2D9GGv2g2SkMX2tuCpBgCPeguzZpyhGrVSIOOgpRz+MjtmZ+g0wG9lstrU6fY+nPPlNdF/UDg";
     const WRONG_SIGNATURE: &str =
         "dQlFxqMQh0idWk6anOerf7b9/XssKkvSrVIv9gMuf7M31ivli6BM2ktCsv9FHB/2FfdwO4LS8muOkFjSt7uAAg";
     const EXPECTED_KEYPAIR_BYTES: [u8; 64] = [
@@ -171,5 +171,29 @@ mod tests {
         let signature = my_keypair.sign(&payload_js).unwrap();
 
         assert_ne!(signature, WRONG_SIGNATURE);
+    }
+
+    #[wasm_bindgen_test]
+    fn pass_incorrect_payload() {
+        #[derive(Serialize, Deserialize, Debug)]
+        struct PayloadErr {
+            method: String,
+            uri: String
+        }
+        let payload_err = PayloadErr {
+            method: "get".to_string(),
+            uri: "/someuri".to_string()
+        };
+
+        let payload_err_js = JsValue::from_serde(&payload_err).unwrap();
+        let my_keypair = HpAdminKeypair::new(
+            HC_PUBLIC_KEY.to_string(),
+            EMAIL.to_string(),
+            PASSWORD.to_string(),
+        )
+        .unwrap();
+        let error = my_keypair.sign(&payload_err_js);
+
+        assert!(error.is_err());
     }
 }

--- a/client/src/lib.rs
+++ b/client/src/lib.rs
@@ -13,7 +13,7 @@ pub struct HpAdminKeypair(Keypair);
 struct Payload {
     method: String,
     uri: String,
-    body_string: String,
+    body: String,
 }
 
 #[wasm_bindgen]
@@ -27,7 +27,7 @@ impl HpAdminKeypair {
         email: String,
         password: String,
     ) -> Fallible<HpAdminKeypair> {
-        console_error_panic_hook::set_once();
+        // console_error_panic_hook::set_once();
         let keypair = new_inner(hc_public_key_string, email, password)?;
         Ok(Self(keypair))
     }
@@ -37,14 +37,14 @@ impl HpAdminKeypair {
     /// const payload = {
     ///     method: String,
     ///     uri: String,
-    ///     body_string: String
+    ///     body: String
     /// }
     /// @example
     /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
     /// const payload = {
     ///     method: "get",
     ///     uri: "/someuri",
-    ///     body_string: ""
+    ///     body: ""
     /// }
     /// myKeys.sign( payload );
     #[wasm_bindgen]
@@ -139,7 +139,7 @@ mod tests {
         let payload = Payload {
             method: "get".to_string(),
             uri: "/someuri".to_string(),
-            body_string: "".to_string(),
+            body: "".to_string(),
         };
 
         let payload_js = JsValue::from_serde(&payload).unwrap();
@@ -159,7 +159,7 @@ mod tests {
         let payload = Payload {
             method: "get".to_string(),
             uri: "/someuri".to_string(),
-            body_string: "".to_string(),
+            body: "".to_string(),
         };
         let payload_js = JsValue::from_serde(&payload).unwrap();
         let my_keypair = HpAdminKeypair::new(

--- a/client/tests/www/index.js
+++ b/client/tests/www/index.js
@@ -1,4 +1,4 @@
-import { HpAdminKeypair } from "@holo-host/hp-admin-key-manager";
+import { HpAdminKeypair } from "../../pkg/hp_admin_key_manager";
 
 const HC_PUBLIC_KEY = "3llrdmlase6xwo9drzs6qpze40hgaucyf7g8xpjze6dz32s957";
 const EMAIL = "pj@abba.pl";
@@ -8,8 +8,8 @@ let kp = new HpAdminKeypair(HC_PUBLIC_KEY, EMAIL, PASSWORD);
 
 const payload = {
     method: "get",
-    uri: "/someuri",
-    body_string: ""
+    uri: 1,
+    body: ""
 }
 
 console.log(kp.sign(payload));

--- a/server/src/main.rs
+++ b/server/src/main.rs
@@ -29,7 +29,7 @@ lazy_static! {
 struct Payload {
     method: String,
     uri: String,
-    body_string: String,
+    body: String,
 }
 
 // Create response based on the request parameters
@@ -62,7 +62,7 @@ fn create_response(req: Request<Body>) -> impl Future<Item = Response<Body>, Err
                     req_uri_string
                 );
 
-                let body_string = match String::from_utf8(body.to_vec()) {
+                let body = match String::from_utf8(body.to_vec()) {
                     Ok(s) => s,
                     Err(e) => {
                         debug!("Error parsing request body: {}", e);
@@ -73,7 +73,7 @@ fn create_response(req: Request<Body>) -> impl Future<Item = Response<Body>, Err
                 let payload = Payload {
                     method: parts.method.to_string(),
                     uri: req_uri_string,
-                    body_string: body_string,
+                    body: body,
                 };
 
                 let public_key = match read_hp_pubkey() {
@@ -220,7 +220,7 @@ mod tests {
         let payload = Payload {
             method: "get".to_string(),
             uri: "/abba".to_string(),
-            body_string: "\"something\": \"interesting\"".to_string(),
+            body: "\"something\": \"interesting\"".to_string(),
         };
 
         let signature = secret_key_exp.sign(&serde_json::to_vec(&payload).unwrap(), &public_key);
@@ -247,7 +247,7 @@ mod tests {
         let payload = Payload {
             method: "get".to_string(),
             uri: "/abba".to_string(),
-            body_string: "\"something\": \"interesting\"".to_string(),
+            body: "\"something\": \"interesting\"".to_string(),
         };
 
         let mut headers = HeaderMap::new();

